src/java.base/unix/classes/sun/nio/fs/UnixSecureDirectoryStream.java

```diff
@@ -118,9 +122,17 @@ private UnixPath getName(Path obj) {
                 ptr = fdopendir(newdfd1);
             } catch (UnixException x) {
                 if (newdfd1 != -1)
-                    UnixNativeDispatcher.close(newdfd1);
+                    try {
+                        UnixNativeDispatcher.close(newdfd1);
+                    } catch (UnixException e) {
+                        x.addSuppressed(e);
```

Comment:
Same here with respect to suppressed exception and what happens at the end of this catch block with `x.rethrowAsIOException(file);`

Reply:
Ditto.